src/ui/components/Library/Team/View/Tests/Overview/TestDetails.tsx

```diff
@@ -1,13 +1,107 @@
-import { TestExecution } from "shared/test-suites/TestRun";
+import orderBy from "lodash/orderBy";
+
+import { GroupedTestRun, TestExecution } from "shared/test-suites/TestRun";
+import MaterialIcon from "ui/components/shared/MaterialIcon";
 import { testFailed, testPassed } from "ui/utils/testRuns";
 
-import { ReplayList } from "./ReplayList";
+import { getTruncatedRelativeDate } from "../../Recordings/RecordingListItem/RecordingListItem";
+import { StatusIcon } from "./StatusIcon";
+
+export function TestDetails({ testRuns }: { testRuns: GroupedTestRun[] }) {
+  const sortedTestRuns = orderBy(testRuns, "date", "desc");
+
+  return (
+    <div className="flex flex-col gap-1 p-2">
+      {sortedTestRuns.map((t: any) => (
+        <TestRun testRun={t} key={t.testRunId} />
+      ))}
+    </div>
+  );
+}
+
+function TestRun({ testRun }: { testRun: GroupedTestRun }) {
+  const executionsWithReplays = testRun.executions.filter(
+    e => e.recordings.length && (testPassed(e) || testFailed(e))
+  );
+
+  if (!executionsWithReplays.length) {
+    return null;
```

Comment:
Under what circumstance would this happen?

Reply:
Test executions past the set retention limit (eg, >7 days) would return the execution, but have no replays attached to it.

For the next batch of work, we're trying to figure out a better model of displaying what should be still-useful data (text execution results) even if it's past the retention limit and doesn't have any replays attached. But for now, this UI/model is replay-centric, so those just get dropped off

Reply:
Could we filter this stuff out from our GraphQL request (so we aren't fetching data to the client that we don't show)?